app/chain_test.go

```diff
@@ -79,3 +79,16 @@ func (ph panicAtHeightDecorator) Deliver(ctx weave.Context, db weave.KVStore, tx
 	}
 	return next.Deliver(ctx, db, tx)
 }
+
+func TestChainNilDecorator(t *testing.T) {
```

Comment:
Can you add a test for cutoffNil?

Just a bit of a table-test...

`[]`
`[nil]`
`[X]`
`[nil, X, Y]`
`[X, Y, nil]`
`[X, nil, Y]`

I think that is all relevant permutations.

Reply:
Done in https://github.com/iov-one/weave/pull/431/commits/7944533a2a4b2a9b81eaec6c44f828a5b331b9fa